tempesta_fw/t/unit/test_sched_rr.c

```diff
@@ -42,102 +42,131 @@
 #include "server.h"
 #include "test.h"
 
-TEST(tfw_sched_rr, sg_empty)
+static TfwMsg *
+sched_rr_get_scheduler_arg (size_t connection_type __attribute__((unused)))
 {
-	int i;
+	return NULL;
+}
 
-	TfwSrvGroup *sg = test_create_sg("test", "round-robin");
+static void
+sched_rr_free_scheduler_arg (TfwMsg * msg __attribute__((unused)))
+{
+	return;
+}
 
-	for (i = 0; i < 3; ++i) {
-		TfwConnection *conn = sg->sched->sched_srv(NULL, sg);
-		EXPECT_TRUE(conn == NULL);
-	}
+static struct TestSchedHelper sched_helper_rr = {
+	.scheduler = "round-robin",
+	.connections_types = 1,
+	.get_scheduler_arg = &sched_rr_get_scheduler_arg,
+	.free_scheduler_arg = &sched_rr_free_scheduler_arg,
+};
 
-	test_sg_release_all();
+TEST(tfw_sched_rr, sg_empty)
+{
+	test_sched_generic_empty_sg(&sched_helper_rr);
 }
 
 TEST(tfw_sched_rr, one_srv_in_sg_and_zero_conn)
 {
-	int i;
-
-	TfwSrvGroup *sg = test_create_sg("test", "round-robin");
-	test_create_srv("127.0.0.1", sg);
-
-	for (i = 0; i < 3; ++i) {
-		TfwConnection *conn = sg->sched->sched_srv(NULL, sg);
-		EXPECT_TRUE(conn == NULL);
-	}
-
-	test_sg_release_all();
+	test_sched_generic_one_srv_in_sg_zero_conn(&sched_helper_rr);
 }
 
+/*
+ * This unit test is implementation aware and checks more than just interface.
+ * Note, that it is very similar to other tests (one_srv_in_sg_and_max_conn and
+ * max_srv_in_sg_and_max_conn) for round-robin and hash schedullers. So if test
+ * structure is changed, other mentioned in above tests should be also be updated
+ */
 TEST(tfw_sched_rr, one_srv_in_sg_and_max_conn)
 {
-	int i;
-	long long s = 0;
+	size_t i, j;
+	long long conn_acc = 0, conn_acc_check = 0;
 
-	TfwSrvGroup *sg = test_create_sg("test", "round-robin");
+	TfwSrvGroup *sg = test_create_sg("test", sched_helper_rr.scheduler);
 	TfwServer *srv = test_create_srv("127.0.0.1", sg);
 
 	for (i = 0; i < TFW_SRV_MAX_CONN; ++i) {
 		TfwSrvConnection *sconn = test_create_conn((TfwPeer *)srv);
 		sg->sched->add_conn(sg, srv, &sconn->conn);
-		s ^= (long long)sconn;
+		conn_acc ^= (long long)&sconn->conn;
+		/* connection reference count must be >0
```

Comment:
The first line of a multi-line comment should contain just a single `/*`. Then please start the sentence with a capital character.